libraries/standard/mqtt/integration-test/mqtt_system_test.c

```diff
@@ -1312,3 +1385,64 @@ void test_MQTT_Restore_Session_Duplicate_Incoming_Publish_Qos2( void )
     /* Make sure that the library cleared the record for the incoming QoS 2 PUBLISH packet. */
     TEST_ASSERT_EQUAL( MQTT_PACKET_ID_INVALID, context.incomingPublishRecords[ 0 ].packetId );
 }
+
+/**
+ * @brief Verifies that the library supports notifying the broker to retain a PUBLISH message
+ * for a topic using the retain flag.
+ */
+void test_MQTT_Publish_With_Retain_Flag( void )
+{
+    /* Publish to a topic with the "retain" flag set. */
+    TEST_ASSERT_EQUAL( MQTTSuccess, publishToTopic( &context,
+                                                    TEST_MQTT_TOPIC,
+                                                    true,  /* setRetainFlag */
+                                                    false, /* isDuplicate */
+                                                    MQTTQoS1,
+                                                    MQTT_GetPacketId( &context ) ) );
+    /* Complete the QoS 1 PUBLISH operation. */
+    TEST_ASSERT_EQUAL( MQTTSuccess,
+                       MQTT_ProcessLoop( &context, 2 * MQTT_PROCESS_LOOP_TIMEOUT_MS ) );
```

Comment:
Should we assert that we received the puback?

Reply:
Good catch